GHzDACs/Cleanup/dac.py

```diff
@@ -936,8 +1024,534 @@ class DacRunner_Build11(DacRunner_Build7):
 class DAC_Build11(DAC_Build7):
     RUNNER_CLASS = DacRunner_Build11
 
+
 fpga.REGISTRY[('DAC', 11)] = DAC_Build11
 
+
+class DacRunner_Build12(DacRunner_Build8):
+    pass
+
+
+class DAC_Build12(DAC_Build8):
+    pass
+
+
+fpga.REGISTRY[('DAC', 12)] = DAC_Build12
+
+
+# Jump table #
+
+class DacRunner_Build15(DacRunner_Build7):
+    # noinspection PyMissingConstructor
+    def __init__(self, dev, reps, start_delay, jt_entries, jt_counters, sram, loop_delay):
+        """ Make a DAC runner for a given device.
+
+        See DAC_Build15.make_jump_table for info on the jt_entries and
+        jt_counters arguments.
+
+        :param DAC_Build15 dev: the device
+        :param reps: number of times to run the board
+        :param start_delay: start delay for this board
+        :param list[(str, list[int])] jt_entries: JT entries
+        :param list[int] jt_counters: JT counter values, or None for all 0s
+        :param str sram: string data to be loaded to sram
+        :return:
+        """
+        self.dev = dev
+        self.reps = reps
+        self.start_delay = start_delay
+        self.loop_delay = loop_delay
+        self.jt_entries = jt_entries
+        self.jt_counters = jt_counters
+        self.sram = sram
+        self.nPackets = 0  # we don't expect any packets back
+        self.seqTime = fpga.TIMEOUT_FACTOR * (100 * self.reps) * 10**-6 + 1  # TODO: what should we do here? issue #49
+
+    def pageable(self):
+        return False  # no paging for JT
+
+    def loadPacket(self, page, isMaster):
+        """ Create pipelined load packet.  For DAC, upload JT and SRAM.
+
+        Note that this add 2 us to the delay for the master board.
+
+        :return: packet for the direct ethernet server
+        """
+        if isMaster:
+            # TODO: how can we add a delay to the JT?
```

Comment:
I was confused at the time because the cycle delay and start delay bytes in the register packet got moved around (and I think renamed in John's doc). But I will delete this TODO.